### PR TITLE
chore: update renovate config

### LINKED ISSUE #5768: Action Required: Fix Renovate Configuration
There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

Location: `.github/renovate.json`
Error type: The renovate configuration file contains some invalid settings
Message: `Invalid configuration option: internalChecksFilters, Invalid configuration option: packageRules[0].manager, Invalid configuration option: packageRules[1].manager, matchUpdateTypes: matchUpdateTypes should be inside a 'packageRule' only, packageRules[0]: Each packageRule must contain at least one match* or exclude* selector. Rule: {"addLabels":["github-actions"],"manager":"github-actions"}, packageRules[1]: Each packageRule must contain at least one match* or exclude* selector. Rule: {"addLabels":["javascript"],"manager":"npm"}`

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## PR Checklist
- [x] Addresses an existing open issue: fixes #5768 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

- Update `renovate.json` to include the config options in correct format

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
@JoshuaKGoldberg The [`matchUpdatesTypes`](https://docs.renovatebot.com/configuration-options/#matchupdatetypes) option doesn't serve any purpose unless paired with some other option like `labels` or other.

**Comment by RahulGautamSingh** [comment]
Tested it this timeon a  [minimal reproduction repo](https://github.com/RahulGautamSingh-testing/ts-eslint)😅

File: .github/renovate.json [hunk 1]
```diff
@@ -2,19 +2,34 @@
...
+      "addLabels": ["pin"]
+    },
+    {
+      "matchUpdateTypes": ["patch"],
+      "addLabels": ["patch"]
+    },
+    {
+      "matchUpdateTypes": ["digest"],
+      "addLabels": ["digest"]
     }
```
Comment (JoshuaKGoldberg) [inline .github/renovate.json:31]: We don't do this today -and didn't filter on version upgrade type previously- so I feel like it's fine to just ignore them now.

```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, let's go with this. Thanks!

File: .github/renovate.json (hunk 1 above)
Comment (JoshuaKGoldberg) [reply .github/renovate.json:31]: I'll just commit this since it's otherwise good to go 🙂

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by bradzacher — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
Anakin in Star Wars saying "It's working, it's working!!"


Screen Shot of dozens of Renovate PRs

**Pull request merged**